### PR TITLE
ioctl: Add debugging feature to show command outputs

### PR DESCRIPTION
**Pull request by ikegami-t**

### CODE REVIEW

File: src/libnvme.map [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # SPDX-License-Identifier: LGPL-2.1-or-later
 
+LIBNVME_1_6 {
```
Comment (tbzatek) [inline src/libnvme.map:3]: The current git master is still v1.5 (unreleased)

Comment (ikegami-t) [reply src/libnvme.map]: Thansk for your comment. Just fixed the patch to use the map file v1.5 definition.

File: src/nvme/tree.h [hunk 2]
```diff
@@ -801,7 +801,7 @@ const char *nvme_ctrl_get_firmware(nvme_ctrl_t c);
 const char *nvme_ctrl_get_model(nvme_ctrl_t c);
 
 /**
- * nvme_ctrl_get_state() - Running state of an controller
+ * nvme_ctrl_get_state() - Running state of a controller
```
Comment (igaw) [inline src/nvme/tree.h:804]: This seems to be an unrelated change.

**Review by igaw — CHANGES_REQUESTED** [review]

File: src/nvme/tree.h (hunk 2 above)
Comment (ikegami-t) [reply src/nvme/tree.h:804]: Splited the change.

File: src/nvme/ioctl.c [hunk 3]
```diff
@@ -86,10 +88,57 @@ static int nvme_submit_passthru64(int fd, unsigned long ioctl_cmd,
...
+	struct timeval end;
+	int err;
+
+	gettimeofday(&start, NULL);
+
+	err = ioctl(fd, ioctl_cmd, cmd);
+
+	gettimeofday(&end, NULL);
+
+	if (nvme_debug(false))
```
[the remark this answers is not in the document]
Comment (ikegami-t) [reply src/nvme/ioctl.c]: Fixed as mentioned.

File: src/nvme/ioctl.c [hunk 4]
```diff
@@ -86,10 +88,57 @@ static int nvme_submit_passthru64(int fd, unsigned long ioctl_cmd,
...
+
 static int nvme_submit_passthru(int fd, unsigned long ioctl_cmd,
 				struct nvme_passthru_cmd *cmd, __u32 *result)
 {
-	int err = ioctl(fd, ioctl_cmd, cmd);
+	struct timeval start;
+	struct timeval end;
+	int err;
+
+	gettimeofday(&start, NULL);
```
[the remark this answers is not in the document]
Comment (ikegami-t) [reply src/nvme/ioctl.c]: This also fixed as suggested. Thank you.

**Comment by igaw** [comment]
I was pondering if this could also be done with ftrace, but I think it is not so easy as I thought.

**Comment by ikegami-t** [comment]
Just rebased.

File: src/libnvme.map [hunk 5]
```diff
@@ -20,6 +20,8 @@ LIBNVME_1_5 {
 		nvme_root_set_application;
 		nvme_subsystem_get_application;
 		nvme_subsystem_set_application;
+		nvme_set_debug;
+		nvme_get_debug;
```
Comment (igaw) [inline src/libnvme.map:24]: These two entries should be under the 1.6 section as we already released 1.5.

File: src/nvme/ioctl.h [hunk 6]
```diff
@@ -3855,4 +3855,16 @@ int nvme_zns_append(struct nvme_zns_append_args *args);
  */
 int nvme_dim_send(struct nvme_dim_args *args);
 
+/**
+ * nvme_debug - Set NVMe command debugging output
```
Comment (igaw) [inline src/nvme/ioctl.h:3859]: kdoc is not happy because `nvme_debug` should be `nvme_set_debug`. The same problem with the kdoc section below

File: src/libnvme.map (hunk 5 above)
Comment (ikegami-t) [reply src/libnvme.map]: Fixed the version.

File: src/nvme/ioctl.h (hunk 6 above)
Comment (ikegami-t) [reply src/nvme/ioctl.h]: Fixed the kdoc also. Thank you.

**Comment by igaw** [comment]
Thanks
